src/controllers/bookings.cr

```diff
@@ -424,6 +440,23 @@ class Bookings < Application
     query.to_a
   end
 
+  private def check_concurrent_bookings(new_booking)
+    # check for concurrent bookings
+    starting = new_booking.booking_start
+    ending = new_booking.booking_end
+    booking_type = new_booking.booking_type
+    user_id = new_booking.user_id_column.defined? ? new_booking.user_id : new_booking.booked_by_id
+
+    query = Booking.query
+      .by_tenant(tenant.id)
+      .where(
+        "booking_start < :ending AND booking_end > :starting AND booking_type = :booking_type AND user_id = :user_id AND rejected = FALSE AND deleted <> TRUE",
+        starting: starting, ending: ending, booking_type: booking_type, user_id: user_id
+      )
+    query = query.where { id != new_booking.id } if new_booking.id_column.defined?
+    query.to_a
```

Comment:
Looks like you only need size of the query from this function? If so I would just return
`query.to_a.size` and change the function name to something like `no_of_concurrent_bookings`

Reply:
I do return the first conflicting booking to mirror the behaviour of the "clashing booking" check.